src/__tests__/__actions__/registration.actions.test.js

```diff
@@ -0,0 +1,179 @@
+import thunk from "redux-thunk";
+import fetchMock from "fetch-mock";
+import configureMockStore from "redux-mock-store";
+import * as actions from "../../redux/actions/registration.action";
+import * as actionTypes from "../../redux/constants/registration.constant";
+import {
+  handleRegistration,
+  handleInputChange,
+  handleBlur
+} from "../../redux/action-creators/registration.action-creator";
+
+const dummyUser = {
+  email: "test@test.com",
+  username: "test",
+  password: "9874",
+  confirmPassword: "9874"
+};
+
+const initialState = {
+  user: {
+    email: "",
+    username: "",
+    password: ""
+  },
+  submitted: false,
+  message: "",
+  success: false
+};
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+describe("test actions ", () => {
+  it("should create an action to request authentication", () => {
+    const expectedAction = {
+      type: actionTypes.REGISTER_REQUESTED,
+      user: dummyUser
+    };
+    expect(actions.requested(dummyUser)).toEqual(expectedAction);
+  });
+
+  it("should create an action for loading", () => {
+    const expectedAction = {
+      type: actionTypes.REGISTER_REQUEST_LOADING
+    };
+    expect(actions.loading()).toEqual(expectedAction);
+  });
+
+  it("should create an action for a successful", () => {
+    const expectedAction = {
+      type: actionTypes.REGISTER_REQUEST_SUCCEEDED,
+      payload: dummyUser
+    };
+    expect(actions.success(dummyUser)).toEqual(expectedAction);
+  });
+  it("should create an action for failure ", () => {
+    const message = "Invalid email";
+    const expectedAction = {
+      type: actionTypes.REGISTER_REQUEST_FAILED,
+      message
+    };
+    expect(actions.failure(message)).toEqual(expectedAction);
+  });
+
+  it("should return an action on input change", () => {
+    const field = "email";
+    const value = "esp.com";
+    const expectedAction = {
+      type: actionTypes.REGISTER_INPUT_CHANGE,
+      payload: { [field]: value }
+    };
+    expect(actions.InputChange(field, value)).toEqual(expectedAction);
+  });
+
+  it("should return an action on blur", () => {
+    Array.of("email", "password", "usermane").forEach(field => {
+      let message;
+      if (field === "password") {
+        message =
+          "The password should be an alphanumeric with at least 8 characters lower and upper cases";
+      } else {
+        message = `Invalid ${field}`;
+      }
+
+      const expectedAction = {
+        type: actionTypes.REGISTER_VALIDATE_INPUT,
+        message
+      };
+      expect(actions.validateInput(message)).toEqual(expectedAction);
+    });
+  });
+});
+
+describe("test action creators", () => {
+  afterEach(() => {
+    fetchMock.restore();
+  });
+  it("fetch successful post to user api call", () => {
+    fetchMock.post("https://titan-devs.herokuapp.com/api/v1/users", {
```

Comment:
consider using development instead of production URL

Reply:
What if I want to call the production API?

Reply:
But we are in development, do you think it is good to use production database in development env?

Reply:
+1 to changing the production to another environment while testing

Reply:
What if I'm a front end developer whose job is to consume the hosted API?
With the development URI, I will be tempted to go and change something in the backend API. which is not good for the long term.
@abayo-luc, we are front end development, but not in backend development...

Reply:
@espoirMur, I think it should be better for the whole team to dynamically load the server URL according to the environment. What is your professional opinion about that?

Reply:
You are right @Musigwa ...
 I’m inclined to agree.

Reply:
Done @Musigwa, I have refactored the code.....